KoreToK: add case for append

SchmErik opened the pull request:
This is similar to ksequence

Fixes: #2245

daejunpark commented on kernel/src/main/java/org/kframework/kompile/KompileFrontEnd.java:
```diff
@@ -50,15 +53,25 @@ public int run() {
 
+    public void saveModuleDerivedParser(ParsedDefinitionWrapper wrapper, String moduleName, KExceptionManager kem) {
+        UserParser parser = wrapper.getModuleDerviedParser(moduleName, kem);
+        String fileName = "extras/" + moduleName +"_Parser.bin";
```
use FileUtil's constant for `extras/`

daejunpark commented on kernel/src/main/java/org/kframework/parser/UserParser.java:
```diff
@@ -0,0 +1,20 @@
+     * @param startSymbol the sort of the program start symbol (non-terminal)
+     * @return an object of ParseResult which includes the K ast and the warnings during the parsing.
+     */
+    public ParseResult parse(String toParse, Source fromSource, Sort startSymbol);
```
cut dependency to kore, by making source and sort be string

daejunpark commented on kernel/src/main/java/org/kframework/krun/KRunOptions.java:
```diff
@@ -61,7 +61,9 @@ public String parser(String mainModuleName) {
                 if (term()) {
                     return kastBinary + " -m " + mainModuleName;
```
add todo

daejunpark commented on kernel/src/main/java/org/kframework/krun/KRun.java:
```diff
@@ -421,9 +425,24 @@ private static String unparseTerm(K input, Module test) {
+            String toParse = FileUtil.read(files.readFromWorkingDirectory(value));
+            String defaultParser = compiledDef.mainSyntaxModuleName() + "_Parser";
+            return parse(toParse, source, startSymbol, defaultParser, files);
         } else {
```
add todo

daejunpark commented on kernel/src/main/java/org/kframework/kompile/ParsedDefinitionWrapper.java:
```diff
@@ -0,0 +1,69 @@
+     * It automatically generates this module unless the user has already defined a module postfixed with
+     * {@link RuleGrammarGenerator#POSTFIX}. In latter case, it uses the user-defined module.
+     */
+    public Option<Module> programParsingModuleFor(String moduleName, KExceptionManager kem) {
```
add comment where this came from, and todo that will remove the original.

daejunpark commented on kernel/src/main/java/org/kframework/kompile/Kompile.java:
```diff
@@ -110,7 +110,7 @@ public CompiledDefinition run(File definitionFile, String mainModuleName, String
 
     /**
-     * Executes the Kompile tool. This tool accesses a
+     * Executes the Kompile tool. This tool first parses the definition and then compiles it.
```
typo?

daejunpark commented on kernel/src/main/java/org/kframework/parser/UserParser.java:
```diff
@@ -0,0 +1,17 @@
+public interface UserParser extends Serializable {
+    /**
+     * @param toParse the text to be parsed
+     * @param fromSource the source of the String toParse
```
Is there any format for the string representation of the source information? Or, is it just a file path?

daejunpark commented on kernel/src/main/java/org/kframework/krun/KRun.java:
```diff
@@ -421,10 +426,28 @@ private static String unparseTerm(K input, Module test) {
 
+
+    public K parse(String toParse, Source source, Sort startSymbol, String moduleDerivedParser, FileUtil files) {
+        String modulePath = "extras/" + moduleDerivedParser + ".bin";
```
this still uses the constant.

daejunpark replied:
You can use the uniform way of getting the file name here as well.

daejunpark commented on kernel/src/main/java/org/kframework/kompile/KompileFrontEnd.java:
```diff
@@ -50,15 +53,25 @@ public int run() {
 
+    public void saveModuleDerivedParser(ParsedDefinitionWrapper wrapper, String moduleName, KExceptionManager kem) {
+        UserParser parser = wrapper.getModuleDerviedParser(moduleName, kem);
+        String fileName = FileUtil.EXTRAS_FOLDER + moduleName +"_Parser.bin";
```
FileUtil may provide a function that takes the module name and returns the file name, and you may want to use it uniformly.

Commit:
KoreToK: add case for append

This is similar to ksequence

Commit:
regression-new: add test case with additional flag

The option `--depth 0` should be removed once the regression in
kore-exec has been resolved. This is a simple test case that exercises
the append case added in KoreToK.

SchmErik commented:
this was make from a fork. I've created another PR from the k repo here https://github.com/kframework/k/pull/2266
